src/main/java/net/sf/jabref/gui/maintable/MainTableSelectionListener.java

```diff
@@ -407,6 +408,9 @@ private void showIconRightClickMenu(MouseEvent e, int row, MainTableColumn colum
                             }
                             menu.add(new ExternalFileMenuItem(panel.frame(), entry, content.get(), content.get(), icon,
                                     panel.getBibDatabaseContext(), field));
+                            if (field == FieldName.DOI) {
```

Comment:
Beware! Although FieldName.XXX looks like an enum, is is String constant, so I think you should better use equals